### PR TITLE
Remove use of V1 wrapper: PassPhrase

### PR DESCRIPTION
**Pull request by denisshevchenko**
# Linked Issue

<Put here a reference to the issue this PR relates to and which requirements it tackles>

<p align="right">#119</p>

# Overview

<Detail in a few bullet points the work accomplished in this PR>

1. - [x] Type `Core.PassPhrase` doesn't use `V1` wrapper.
2. - [x] newtype `WalletPassPhrase` is introduced instead.

# Comments

<Additional comments or screenshots to attach if any>

¯\\\_(ツ)\_/¯

# Checklist

- [x] I've kept this PR simple, on-the-spot, free of cosmetic changes.
- [x] I have reviewed my commit messages and history.
- [x] I've assigned a reviewer.
- [x] I acknowledge my changes may require an update in the wiki.
- [x] I have added a reference to this PR to the corresponding issue.

---

- [x] I've checked the checklist

### CODE REVIEW

File: src/Cardano/Wallet/API/V1/Types.hs [hunk 1]
```diff
@@ -237,21 +236,18 @@ mkPassPhrase text =
...
         Left e    -> fail (toString e)
-        Right pp' -> pure (V1 pp')
+        Right pp' -> pure pp'
     parseJSON x           = typeMismatch "parseJSON failed for PassPhrase" x
 
-instance Arbitrary (V1 Core.PassPhrase) where
-    arbitrary = fmap V1 arbitrary
-
-instance ToSchema (V1 Core.PassPhrase) where
+instance ToSchema Core.PassPhrase where
```
Comment (KtorZ) [inline src/Cardano/Wallet/API/V1/Types.hs]: Here we introduce orphan instances, which we really don't want. Instead, let's come up with our own dedicated newtype: `PassPhrase` or `WalletPassPhrase`. The idea isn't to simply remove `V1` but to replace it with appropriated newtypes!

**Review by akegalj — APPROVED** [review]

**Review by KtorZ — CHANGES_REQUESTED** [review]
PR looks good, but we're missing two things:

- Golden tests
- Acid-State Migration

Ideally, we need write golden tests before making any change, such that they are written using the old code, and we control via the new code that everything is correct. 
Can you maybe issue another PR that we'll merge prior to this one, only adding at least one golden for the `ToJSON` and `FromJSON` instance, even though I am quite sure nothing got broken along the way.

But for all others, that should become the normal procedure:

1/ Golden tests
2/ Removing / Replacing V1
3/ Acid-state Migration.

**Comment by KtorZ** [comment]
@denisshevchenko 

Remember to update golden tests here as a consequence of #144  :+1:

File: src/Cardano/Wallet/API/V1/Types.hs [hunk 2]
```diff
@@ -343,20 +338,30 @@ instance ToSchema (V1 Core.Timestamp) where
...
 
-instance Semigroup (V1 Core.PassPhrase) where
-    V1 a <> V1 b = V1 (a <> b)
+instance Semigroup WalletPassPhrase where
+    WalletPassPhrase a <> WalletPassPhrase b = WalletPassPhrase (a <> b)
 
-instance Monoid (V1 Core.PassPhrase) where
-    mempty = V1 mempty
+instance Monoid WalletPassPhrase where
+    mempty = WalletPassPhrase mempty
```
Comment (KtorZ) [inline src/Cardano/Wallet/API/V1/Types.hs:347]: Let's avoid defining instances on type aliases. Could we actually remove the "SpendingPassword" alias entirely (while preserving the existing fields in various data types?)

**Review by KtorZ — APPROVED** [review]
LGTM :+1:

One minor gotcha that we could fix as part of another PR I believe.

**Pull request merged**